packages/rum-core/src/domain/action/getActionNameFromElement.ts

```diff
@@ -1,26 +1,64 @@
 import { safeTruncate, isIE, find } from '@datadog/browser-core'
 import { getParentElement } from '../../browser/polyfills'
+import { NodePrivacyLevel, PRIVACY_ATTR_NAME } from '../../constants'
+import { getNodePrivacyLevel } from '../privacy'
 
 /**
  * Get the action name from the attribute 'data-dd-action-name' on the element or any of its parent.
  * It can also be retrieved from a user defined attribute.
  */
 export const DEFAULT_PROGRAMMATIC_ACTION_NAME_ATTRIBUTE = 'data-dd-action-name'
+export const ACTION_NAME_PLACEHOLDER = 'Masked Element'
+export function getActionNameFromElement(
+  element: Element,
+  privacyEnabledForActionName?: boolean,
+  userProgrammaticAttribute?: string
+): { name: string; masked?: boolean } {
+  // to determine if we should add placeholder
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏The implementation of the function could probably be simplified like so (to be testes :)
```ts
const userDefineActionName =
    getActionNameFromElementProgrammatically(element, DEFAULT_PROGRAMMATIC_ACTION_NAME_ATTRIBUTE) ||
    (userProgrammaticAttribute && getActionNameFromElementProgrammatically(element, userProgrammaticAttribute))
  let automaticActionName: string | undefined
  if (!privacyEnabledForActionName && !userDefineActionName) {
    automaticActionName =
      getActionNameFromElementForStrategies(
        element,
        userProgrammaticAttribute,
        priorityStrategies,
        privacyEnabledForActionName
      ) ||
      getActionNameFromElementForStrategies(
        element,
        userProgrammaticAttribute,
        fallbackStrategies,
        privacyEnabledForActionName
      )
  }
  const placeholder = privacyEnabledForActionName ? PRIVATE_ACTION_NAME_PLACEHOLDER : ''

  return {
    name: userDefineActionName || automaticActionName || placeholder,
    masked: privacyEnabledForActionName && !userDefineActionName,
  }
```